arxiv/metadata/metacheck.py

```diff
@@ -59,10 +59,51 @@ def contains_outside_math(s1:str, s2:str) -> bool:
         return False
        
 
+############################################################
+# Check report objects are a string (disposition)
+# and a list of complaints
+# Each complaint is a string and a list of contexts.
+
+class MetadataCheckReport:
+    def __init__(self):
+        self.disposition = OK
+        # map from complaint to context (may be "")
+        self.complaints = {}
+
+    def __repr__(self):
+        if self.disposition == OK:
+            return f"<MetadataCheckReport OK>"
+        else:
+            return f"<MetadataCheckReport {self.disposition}: {self.get_complaints_strings()}>"
+        
+
+    def add_complaint(self, disposition, complaint_string:str, context:str = ""):
+        self.disposition = combine_dispositions(self.disposition, disposition)
+        if complaint_string in self.complaints:
+            self.complaints[complaint_string].append(context)
+        else:
+            self.complaints[complaint_string] = [context]
+        #

```

Comment:
Could we use a `set` here to not do the inclusion checks?

Reply:
No. The first check is necessary in order to accumulate a list of contexts; I don't think I want a set. 

I can use a defaultdict(list) if you like.